File: dev/test-client.pl
use IO::Socket::INET;

# from http://xmodulo.com/how-to-write-simple-tcp-server-and-client-in-perl.html 

# auto-flush on socket
$| = 1;
 
# create a connecting socket
my $socket = new IO::Socket::INET (
    PeerHost => 'localhost',
    PeerPort => '15555',
    Proto => 'tcp',
    );
die "cannot connect to the server $!\n" unless $socket;
print "connected to the server\n";



# data to send to a server
# my $req = "hallo\nCLASSIFIER=cld2\nLANGHINT=de\n";
my $req = "hallo\nCLASSIFIER=cld2\n";
my $size = $socket->send($req);
print "sent data of length $size\n";
$socket->send('<<<CLASSIFY>>>');

# notify server that request has been sent
shutdown($socket, 1);
 
# receive a response of up to 1024 characters from server
my $response = "";
$socket->recv($response, 1024);
print "received response: $response\n";
 
$socket->close();

File: dev/test-client.py
import asyncio, socket

request = None

try:
    while request != 'quit':
        request = input('>> ')
        if request:
            server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
            server.connect(('localhost', 15555))
            request += "\n"
            server.send(request.encode('utf8'))
            command = "<<CLASSIFY>>\n"
            server.send(command.encode('utf8'))
            response = server.recv(255).decode('utf8')
            print(response)
            server.close()

            server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
            server.connect(('localhost', 15555))
            server.send(request.encode('utf8'))
            command = "CLASSIFIER=cld2\n"
            server.send(command.encode('utf8'))
            command = "<<CLASSIFY>>\n"
            server.send(command.encode('utf8'))
            response = server.recv(255).decode('utf8')
            print("cld2: " + response)
            server.close()

            server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
            server.connect(('localhost', 15555))
            server.send(request.encode('utf8'))
            command = "CLASSIFIER=cld2\n"
            server.send(command.encode('utf8'))
            command = "LANGHINT=de\n"
            server.send(command.encode('utf8'))
            command = "<<CLASSIFY>>\n"
            server.send(command.encode('utf8'))
            response = server.recv(255).decode('utf8')
            print("cld2 + langhint: " + response)
            server.close()

except KeyboardInterrupt:
    server.close()
